modules/client-common/src/utils.ts

```diff
@@ -405,29 +423,46 @@ export async function* prepareGenericUpdate(
  */
 export function getNetwork(networkish: Networkish): Network {
   let network: Network | undefined;
-  for (const nw of ADDITIONAL_NETWORKS) {
+  try {
+    network = ethersGetNetwork(networkish);
+  } catch {
+  } finally {
     switch (typeof networkish) {
       case "string":
-        if (networkish === nw.name) {
-          network = nw;
-        }
-        break;
       case "number":
-        if (networkish === nw.chainId) {
-          network = nw;
+        const aragonNw = getNetworkByNameAliasOrChainId(networkish);
+        if (!aragonNw) {
+          throw new UnsupportedNetworkError(networkish.toString());
         }
+        const ethers5Alias = getNetworkAlias(
+          SupportedAliases.ETHERS_5,
+          aragonNw.name,
+        );
+        const nwDeployment = getNetworkDeploymentForVersion(
+          aragonNw.name,
+          SupportedVersions.V1_3_0,
+        );
+        if (!nwDeployment) {
+          throw new UnsupportedNetworkError(aragonNw.name);
+        }
+        const ensRegistryAddress = nwDeployment.ENSRegistry?.address;
+        network = {
+          name: ethers5Alias || aragonNw.name,
+          chainId: aragonNw.chainId,
+          ensAddress: ensRegistryAddress,
+        };
         break;
       case "object":
-        if (networkish.name === nw.name && networkish.chainId === nw.chainId) {
-          network = nw;
+        if (networkish.name && networkish.chainId) {
+          break;
         }
         break;
```

Comment:
what is this for?

Reply:
forgot to add the `network = networkish` inside